core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -98,6 +99,9 @@ case class SparkListenerExecutorAdded(time: Long, executorId: String, executorIn
 case class SparkListenerExecutorRemoved(time: Long, executorId: String, reason: String)
   extends SparkListenerEvent
 
+@DeveloperApi
+case class SparkListenerBlockUpdated(updateBlockInfo: UpdateBlockInfo) extends SparkListenerEvent
```

Comment:
Right. Just added a new develop class `BlockUpdatedInfo`. And I also updated `JsonProtocol` to support the new `SparkListenerBlockUpdated`.